docs/design/schema-compatibility.md

```diff
@@ -0,0 +1,119 @@
+
+Each version has its distinct schema, state and its own queries/requests. This strategy eliminates the need for data migration and allows different schema versions to coexist seamlessly.
+
+**Strategy steps:**
+
+- For each update:
+  - Create a new version of a Schema and state (a new .proto file)
+  - Implement transactions and queries for the new schema version.
+
+#### Option 2: Generic Schema Storage (Not Recommended for Production)
+
+**Description:**
+
+Implement a flexible, generic schema structure that can support a wide range of data formats.
+
+While offering a robust solution for handling radical changes, this method requires careful planning and development, which can potentially take a significant amount of time.
+
+**Strategy steps:**
+
+- One time actions:
+  - Create a more flexible, generic schema structure to hold a wide range of data formats (Can be used [Any](https://github.com/protocolbuffers/protobuf/blob/main/src/google/protobuf/any.proto) as described in [ADR-19](https://docs.cosmos.network/v0.47/build/architecture/adr-019-protobuf-state-encoding#usage-of-any-to-encode-interfaces))
+  - Migrate old states to the newer, generic schema.
+  - Remove the states associated with the older schema versions.
+  - Optioanlly can be implemented queries for requesting schemas with any return type
+- For each update:
+  - Create a new Schema version (a new .proto file)
+  - Implement transactions and queries that can handle data according to its version, including mechanisms for converting generic values into the corresponding schema version.
+
+## New version replaces the legacy one (V2 replaces V1)
+
+### Strategy for Compatible changes (Not keeping backward compatibility in API)
+
+For changes that are backward-compatible, such as adding optional fields or extending enumerations.
+
+**Description:**
+
+This strategy focuses on updating the schema without ensuring backward compatibility at the API level. Since the schemas are compatible, there will likely be no need for migration.
+
+**Strategy steps:**
+
+- For each update:
+  - Update the schema by introducing compatible changes (such as adding a new optional field).
+  - Migrate old states to the newer if needed.
+  - Update transactions and queries if needed.
+
+### Strategy for Non-Compatible changes (Not keeping backward compatibility in API)
+
+For changes that affect compatibility, like adding mandatory fields.
+
+**Description:**
+
+This strategy focuses on updating the schema without ensuring backward compatibility at the API level. Since the schemas are not compatible, migration is carried out manually through a special transaction.
+
+**Strategy steps:**
+
+- For each update:
+  - Update the schema by introducing changes.
+  - Update transactions and queries if needed.
+  - Add a new transaction to fulfill new required fields (essentially this is a manual migration via transactions)
+
+### Strategy for Non-Compatible changes (Keeping backward compatibility in API)
```

Comment:
It seems this strategy is suitable for compatible (actually convertable) changes only.
So, probably we can rename the Options here a bit:

- Strategy for Compatible or Convertable changes
  - Option 1: Not keeping backward compatibility in API
  - Option 2: Keeping backward compatibility in API
- Strategy for Non-convertable changes
  - Option 1: Not keeping backward compatibility in API
  - Option 2: Keeping backward compatibility in API

I think we can add the following for the last option:
It's not possible to replace an old version here. "Multiple versions can live in parallel: Strategy for Non-Compatible Changes" options should be used instead.